src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
Ooh, that's a good point - I hadn't considered bytes.  IMO the best answer would be to support either depending on the type of the argument, clearly document that both are supported, and to use a name like `from_regex` (unimaginative but obvious).